### ISSUE TITLE
Windows containers cant utilize KUBERNETS_SERVICE_HOST/PORT

### ISSUE THREAD

**Issue by johnSchnake** [kubernetes-sigs/sig-windows-dev-tools#116]
See https://github.com/vmware-tanzu/sonobuoy/issues/1443

TL;DR; kube-proxy is up and running and works for containers on the linux node. Just not windows containers on the windows node.

**Comment by johnSchnake** [comment]
I tried just running it as a service directly on the node but it needs to be a pod so that it has access to the right tokens and such.

**Comment by johnSchnake** [comment]
So all the steps I was running to get kube-proxy up are already in the code but I had tried the 'quick start' which didn't do that.

Now I have kube proxy running and it seems to be working though I have some permission issue to figure out.

Thanks!

**Comment by johnSchnake** [comment]
I'll keep looking into this; need to just see if kube-proxy is running the way it is supposed to be.

In the err logs I see that it fails to find the kubeconfig file (required since its running as a service on the node and not a pod). But it doesn't exist in the place it looks for. I tried to manually set the HOST/PORT env vars before and then it just fails because the in cluster config assumes it will also be able to find tokens for permission in the container its running on.

Not certain this isn't still a kube-proxy configuration issue or just my local setup.

**Comment by johnSchnake** [comment]
```
Because Kubernetes on Windows cannot run Calico in a pod with an auto-provisioned service account, Calico requires a kubeconfig file to access the API server. This section describes how to find an existing calico-node service account used by Calico on Linux side, and then to export the service account token as a kubeconfig file for Calico to use.
```

I have a lot to learn about the windows networking stuff so maybe this is just news to me but I'm just following google and seeing where it takes me...

**Comment by jsturtevant** [comment]
> Because Kubernetes on Windows cannot run Calico in a pod with an auto-provisioned service account

This is because calico and kube components must currently run as Services on windows because we don't have privileged containers.  If these ran as pods they could use the auto-provisioned SAs.  

We have been adding hostprocess containers (available in 1.22 in alpha) which would enable this feature.  I've got an example of how this would work here https://github.com/kubernetes-sigs/sig-windows-tools/pull/161. But it is still pretty early days...
